Tests/ApodiniTests/SemanticModel/EnrichedInfoTests.swift

```diff
@@ -0,0 +1,160 @@
+            propertyInfo: nil,
+            propertiesOffset: nil
+        )
+        
+        XCTAssertEqual(
+            EnrichedInfo.Cardinality.zeroToOne,
+            EnrichedInfo.Cardinality.zeroToOne
+        )
+        XCTAssertEqual(
+            EnrichedInfo.Cardinality.exactlyOne,
+            EnrichedInfo.Cardinality.exactlyOne
+        )
+        XCTAssertEqual(
+            EnrichedInfo.Cardinality.zeroToMany(.array),
+            EnrichedInfo.Cardinality.zeroToMany(.array)
+        )
+        XCTAssertEqual(
+            EnrichedInfo.Cardinality.zeroToMany(
+                .dictionary(
+                    key: keyType,
+                    value: valueType
+                )
+            ),
+            EnrichedInfo.Cardinality.zeroToMany(
+                .dictionary(
+                    key: keyType,
+                    value: valueType
+                )
+            )
+        )
+        XCTAssertNotEqual(
+            EnrichedInfo.Cardinality.zeroToMany(
+                .array
+            ),
+            EnrichedInfo.Cardinality.zeroToMany(
+                .dictionary(
+                    key: keyType,
+                    value: valueType
+                )
+            )
+        )
+        XCTAssertNotEqual(
+            EnrichedInfo.Cardinality.zeroToOne,
+            EnrichedInfo.Cardinality.exactlyOne
+        )
+        XCTAssertNotEqual(
+            EnrichedInfo.Cardinality.zeroToOne,
+            EnrichedInfo.Cardinality.zeroToMany(.array)
+        )
+        XCTAssertNotEqual(
+            EnrichedInfo.Cardinality.exactlyOne,
+            EnrichedInfo.Cardinality.zeroToMany(.array)
+        )
+    }
+
+    func testCollectionContextIsEquatable() throws {
+        let keyType = EnrichedInfo(
+            typeInfo: try typeInfo(of: String.self),
+            propertyInfo: nil,
+            propertiesOffset: nil
+        )
+        let valueType = EnrichedInfo(
+            typeInfo: try typeInfo(of: Int.self),
+            propertyInfo: nil,
+            propertiesOffset: nil
+        )
+        let valueType1 = EnrichedInfo(
+            typeInfo: try typeInfo(of: String.self),
+            propertyInfo: nil,
+            propertiesOffset: nil
+        )
+        
+        XCTAssertEqual(
+            EnrichedInfo.CollectionContext.array,
+            EnrichedInfo.CollectionContext.array
+        )
+        XCTAssertEqual(
+            EnrichedInfo.CollectionContext.dictionary(key: keyType, value: valueType),
+            EnrichedInfo.CollectionContext.dictionary(key: keyType, value: valueType)
+        )
+        XCTAssertNotEqual(
+            EnrichedInfo.CollectionContext.dictionary(key: keyType, value: valueType),
+            EnrichedInfo.CollectionContext.array
+        )
+        XCTAssertNotEqual(
+            EnrichedInfo.CollectionContext.dictionary(key: keyType, value: valueType),
+            EnrichedInfo.CollectionContext.dictionary(key: keyType, value: valueType1)
+        )
+    }
+
```

Comment:
I prototyped a helper function. Maybe you find it useful:
```swift
func testCardinality(_ lhs: Any.Type, _ rhs: Any.Type, isEqual: Bool) {
    let keyPath = \Node<EnrichedInfo>.value.cardinality
    
    if isEqual {
        XCTAssertEqual(
            EnrichedInfo.node(lhs)[keyPath: keyPath],
            EnrichedInfo.node(rhs)[keyPath: keyPath]
        )
    } else {
        XCTAssertNotEqual(
            EnrichedInfo.node(lhs)[keyPath: keyPath],
            EnrichedInfo.node(rhs)[keyPath: keyPath]
        )
    }
}
```

Reply:
Thanks! This would entail to call `.edited(handleArray)` in between, or am I missing something? Otherwise it would always compare two default cardinality of `.exactlyOne`

Reply:
No, it wouldn't call `edited`, even though it was my intention, kind of. 😄
Anyways, my first thought was to avoid initializing `EnrichedInfo` manually.

However, now I am thinking of refactoring/removing this whole `edited` stuff and doing it in `EnrichedInfo.node`. Compare it to now, where we alter the `Tree` afterwards 😅

Reply:
I just refactored the code in e73f561 :D